app/controllers/articles_controller.rb

```diff
@@ -24,4 +24,36 @@ def update
     #WIP
   end
 
+  def publish
+    article = prepare_article(params[:id])
+    article.status = Article::PUBLISHED
+    unless article.save
+      redirect_to admin_categories_path(article.category_key), alert: 'An error while publishing article'
+    end
+    redirect_to admin_categories_path(article.category_key), notice: 'Article published'
+  end
+
+  def unpublish
+    article = prepare_article(params[:id])
+    article.status = Article::UNPUBLISHED
+    unless article.save
+      redirect_to admin_categories_path(article.category_key), alert: 'An error while unpublishing article'
+    end
+    redirect_to admin_categories_path(article.category_key), notice: 'Article unpublished'
+  end
+
+  def destroy
+    article = Article.left_joins(:category).select('articles.*, categories.key as category_key').find(params[:id])
```

Comment:
`Article.left_joins(:category).select('articles.*, categories.key as category_key')` looks obsolete here, what is the reason?